deploy/docker/notebook/base/base.yaml

```diff
@@ -421,26 +425,92 @@ addfiles:
           if "NB_UMASK" in os.environ:
               os.umask(int(os.environ["NB_UMASK"], 8))
 
-          def is_inside_path(path_to_test, path):
-              return os.path.samefile(os.path.commonpath([path, path_to_test]), path)
-        
-          # Configure file post-save hook
-          def script_post_save(model, os_path, contents_manager, **kwargs):
+          # Configure file post-save, post-rename, and post-delete hooks
+          
+          #### Hook Class ####
+
+          class hookVariables():
+              POLUS_NOTEBOOKS_HUB_API = os.environ["POLUS_NOTEBOOKS_HUB_API"]
               work_path = '/home/jovyan/work/'
               shared_path = '/home/jovyan/shared/notebooks'
 
-              if not (is_inside_path(os_path, work_path) or is_inside_path(os_path, shared_path)):
-                    return
+          class HookFileContentsManager(FileContentsManager):
+              """Local spawner that runs single-user servers as the same user as the Hub itself.
+              Overrides user-specific env setup with no-ops.
+              """
 
-              logging.getLogger().info(f"Saved notebook at {os_path}")
-              try:
-                    r = requests.post("https://polus-notebooks-hub-backend.ci.aws.labshare.org/file-data/new-file", json={"user": os.getenv("JUPYTERHUB_USER"), "filePath": os_path}, timeout=1)
-                    if r.status_code == 200:
-                        logging.getLogger().info(f"Registered {os_path} with Notebooks Hub")
-                    else:
-                        logging.getLogger().error(f"Could not register {os_path} with Notebooks Hub. Return code: {r.status_code}")
-              except Exception as e:
-                    logging.getLogger().error(f"Could not register {os_path} with Notebooks Hub", exc_info=e)
+              def is_inside_path(self, path_to_test, path):
+                  return os.path.samefile(os.path.commonpath([path, path_to_test]), path)
               
-          
-          c.FileContentsManager.post_save_hook = script_post_save
\ No newline at end of file
+              def script_post_delete(self, os_path):
+
+                  hookVar = hookVariables()
+
+                  os_path = str(Path().resolve()) + os_path
+                  
+                  if not (self.is_inside_path(os_path, hookVar.work_path) or self.is_inside_path(os_path, hookVar.shared_path)):
+                      return
+                  
+                  try:
+                      r = requests.post(hookVar.POLUS_NOTEBOOKS_HUB_API+"/file-data/delete-file", json={"user": os.getenv("JUPYTERHUB_USER"), "filePath": os_path}, timeout=1)
+                  except Exception as e:
+                      if os.environ["POLUS_NOTEBOOKS_HUB_FILE_LOGGING_ENABLED"] == "True":
+                          logging.getLogger().error(f"Failed to delete {os_path} from Notebooks Hub", exc_info=e)
```

Comment:
Is there a reason why the contents of this function cannot be defined inside of `delete` method. Ditto for save and rename

Reply:
the reason I created it like so is incase there is a future need for a pre-save/rename/delete hook. Currently it operates as a post hook, but I wanted to leave it open to that possible future. And there is more clarity in my opinion as it is. But I can change it if you would like.

Reply:
That makes sense, let's keep it that way